velox/connectors/hive/tests/HivePartitionUtilTest.cpp

```diff
@@ -117,3 +125,51 @@ TEST_F(HivePartitionUtilTest, partitionName) {
         "Unsupported partition type: MAP");
   }
 }
+
+TEST_F(HivePartitionUtilTest, partitionNameForNull) {
+  RowVectorPtr input = makeRowVector(
+      {"flat_bool_col",
+       "flat_tinyint_col",
+       "flat_smallint_col",
+       "flat_int_col",
+       "flat_bigint_col",
+       "dict_string_col",
+       "const_date_col"},
+      {makeNullableFlatVector<bool>(
+           std::vector<std::optional<bool>>{std::nullopt}),
+       makeNullableFlatVector<int8_t>(
+           std::vector<std::optional<int8_t>>{std::nullopt}),
+       makeNullableFlatVector<int16_t>(
+           std::vector<std::optional<int16_t>>{std::nullopt}),
+       makeNullableFlatVector<int32_t>(
+           std::vector<std::optional<int32_t>>{std::nullopt}),
+       makeNullableFlatVector<int64_t>(
+           std::vector<std::optional<int64_t>>{std::nullopt}),
+       makeNullableDictionary<StringView>(
```

Comment:
The usage of makeNullableDictionary here is strange. It seems to create a vector with 10 rows while other columns have only 1 row. Is this intentional? 

I assume makeNullableDictionary function is not needed and can be replaced with makeNullableFlatVector.